devito/passes/iet/orchestration.py

```diff
@@ -72,23 +69,39 @@ def _make_withlock(self, iet, sync_ops, layer):
 
         return iet, [efunc]
 
-    def _make_fetchupdate(self, iet, sync_ops, layer):
-        body, prefix = fetchupdate(layer, iet, sync_ops, self.lang, self.sregistry)
+    def _make_callable(self, name, iet, *args):
+        name = self.sregistry.make_name(prefix=name)
+        efunc = make_callable(name, iet.body)
 
-        # Turn init IET into a Callable
-        name = self.sregistry.make_name(prefix=prefix)
-        body = List(body=body)
-        parameters = derive_parameters(body, ordering='canonical')
-        efunc = Callable(name, body, 'void', parameters, 'static')
-
-        # Perform initial fetch by the main thread
-        iet = List(
-            header=c.Comment("Initialize data stream"),
-            body=Call(name, parameters)
-        )
+        iet = Call(name, efunc.parameters)
 
         return iet, [efunc]
 
+    def _make_initarray(self, iet, *args):
+        return self._make_callable('init_array', iet, *args)
+
+    def _make_snapout(self, iet, *args):
+        return self._make_callable('write_snapshot', iet, *args)
+
+    def _make_snapin(self, iet, *args):
+        return self._make_callable('read_snapshot', iet, *args)
```

Comment:
I would keep the func name as the callable arg for being homogeneous to 'init_array'